Sources/Swiftly/Install.swift

```diff
@@ -59,36 +59,64 @@ struct Install: SwiftlyCommand {
     @Flag(inversion: .prefixedNo, help: "Verify the toolchain's PGP signature before proceeding with installation.")
     var verify = true
 
+    @Option(help: ArgumentHelp(
+        "A file path to a location for a post installation script",
+        discussion: """
+        If the toolchain that is installed has extra post installation steps they they will be
+        written to this file as commands that can be run after the installation.
+        """
+    ))
+    var postInstallFile: String?
+
     private enum CodingKeys: String, CodingKey {
-        case version, token, use, verify
+        case version, token, use, verify, postInstallFile
     }
 
     mutating func run() async throws {
+        try validateSwiftly()
+
         let selector = try ToolchainSelector(parsing: self.version)
         SwiftlyCore.httpClient.githubToken = self.token
         let toolchainVersion = try await self.resolve(selector: selector)
         var config = try Config.load()
-        try await Self.execute(
+        let postInstallScript = try await Self.execute(
             version: toolchainVersion,
             &config,
             useInstalledToolchain: self.use,
             verifySignature: self.verify
         )
+
+        if let postInstallScript = postInstallScript {
```

Comment:
Is it ok to install the dependencies after the swift toolchain has been installed?

Reply:
Yes, I believe so. Swiftly doesn't need to run anything inside the toolchain, it just expands the tarball or macOS pkg. We just need the user to be aware of the dependencies that they don't have that the toolchain needs so that they can take those steps before using it. It's also possible in the future that newer toolchains may have more/less dependencies, or maybe none at all.

Reply:
I guess the install is just unzipping a file in the end, so yeah just ignore this